### PR TITLE
Assertion deleted. Fix for icu-parameter.

### PR DESCRIPTION
**Pull request by alexbakharew**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers, motivation, rationale - **mandatory**)*

- [X] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [X] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.9: *(Please link PR)*
  - [ ] Backport for 3.8: *(Please link PR)*
  - [ ] Backport for 3.7: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: lib/Basics/Utf8Helper.cpp [hunk 1]
```diff
@@ -218,7 +223,7 @@ bool Utf8Helper::setCollatorLanguage(std::string_view lang,
   return true;
 }
 
-std::string Utf8Helper::getCollatorLanguage() {
```
Comment (gnusi) [inline lib/Basics/Utf8Helper.cpp:221]: I do remember we've agreed upon using `getName()` always. At creation time locale has to be adjusted.

**Review by gnusi — CHANGES_REQUESTED** [review]

File: lib/Basics/Utf8Helper.h [hunk 2]
```diff
@@ -125,9 +125,11 @@ class Utf8Helper {
 
   //////////////////////////////////////////////////////////////////////////////
   /// @brief get collator language
+  /// @param langType type of language. Now supports DEFAULT and ICU only
+  /// collation
   //////////////////////////////////////////////////////////////////////////////
 
-  std::string getCollatorLanguage();
+  std::string getCollatorLanguage(LanguageType langType = LanguageType::DEFAULT);
```
Comment (Dronplane) [inline lib/Basics/Utf8Helper.h:132]: As we now return full locale name from this method, maybe it is better to rename it to `getCollactorLocale` and corresponding `setCollatorLanguage` -> `setCollatorLocale`

File: lib/ApplicationFeatures/LanguageFeature.cpp [hunk 3]
```diff
@@ -198,8 +198,10 @@ void LanguageFeature::prepare() {
   auto context = ArangoGlobalContext::CONTEXT;
   std::string binaryExecutionPath = context->getBinaryPath();
   std::string binaryName = context->binaryName();
-  _icuData = LanguageFeature::prepareIcu(_binaryPath, binaryExecutionPath, p,
-                                         binaryName);
+  if (_icuData.empty()) {
```
Comment (Dronplane) [inline lib/ApplicationFeatures/LanguageFeature.cpp:201]: Do we still need this change after fix for tests?

Comment (alexbakharew) [reply lib/ApplicationFeatures/LanguageFeature.cpp:201]: Yes. If we launch 100 tests in a row it leads to high memory consumption

Comment (Dronplane) [reply lib/ApplicationFeatures/LanguageFeature.cpp:201]: But maybe just trigger cleanup of loaded IcuDataPtr inside Utf8Helper in test tear down ?

File: lib/Basics/Utf8Helper.cpp [hunk 4]
```diff
@@ -177,11 +177,19 @@ bool Utf8Helper::setCollatorLanguage(std::string_view lang,
...
     coll = icu::Collator::createInstance(status);
   } else {
-    icu::Locale locale(lang.data());
-    coll = icu::Collator::createInstance(locale, status);
+    icu::Locale canonicalLocale =
+        icu::Locale::createCanonical(std::string(lang).c_str());
+    if (LanguageType::DEFAULT == langType) {
+      std::string l = canonicalLocale.getLanguage();
+      std::string c = canonicalLocale.getCountry();
+      icu::Locale defaultLocale = icu::Locale(l.c_str(), c.c_str());
```
Comment (gnusi) [inline lib/Basics/Utf8Helper.cpp:188]:
```
    icu::Locale defaultLocale{ canonicalLocale.getLanguage(), canonicalLocale.getCountry() };
```

**Review by gnusi — CHANGES_REQUESTED** [review]

**Review by Dronplane — COMMENTED** [review]
LGTM. Waiting for final tests updates

**Review by Dronplane — APPROVED** [review]

**Review by gnusi — APPROVED** [review]

**Pull request merged**